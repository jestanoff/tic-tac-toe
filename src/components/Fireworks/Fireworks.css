.canvas {
    position: absolute;
    z-index: 2;
    left: 0;
    top: 0;
}
