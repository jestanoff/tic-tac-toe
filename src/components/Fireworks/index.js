import React, { memo, useEffect } from 'react';
import PropTypes from 'prop-types';
import { startFireworks } from 'helpers';
import styles from './Fireworks.css';

const Fireworks = ({ handleClick }) => {
  useEffect(() => {
    startFireworks();
    const id = setTimeout(handleClick, 30000);
    return () => clearTimeout(id);
  }, [handleClick]);

  return <canvas id="canvas" className={styles.canvas} onClick={handleClick} />;
};

Fireworks.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default memo(Fireworks);
