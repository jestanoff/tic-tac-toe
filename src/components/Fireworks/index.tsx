import * as React from 'react';
import { startFireworks } from 'helpers';
import styles from './Fireworks.css';

type FireworksProps = {
  handleClick: () => void,
};

const Fireworks = ({ handleClick }) => {
  React.useEffect(() => {
    startFireworks();
    const id = setTimeout(handleClick, 30000);
    return () => clearTimeout(id);
  }, [handleClick]);

  return <canvas id="canvas" className={styles.canvas} onClick={handleClick} />;
};

export default React.memo(Fireworks);
