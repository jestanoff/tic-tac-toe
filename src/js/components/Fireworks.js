import React, { Component, PropTypes } from 'react';
import { canvas } from '../../css/fireworks.css';

class Fireworks extends Component {
    componentDidMount() {
        window.startFireworks();
        setTimeout(this.props.handleClick, 60000);
    }

    render() {
        return <canvas id='canvas' className={ canvas } onClick={ this.props.handleClick } />;
    }
}

Fireworks.propTypes = {
    handleClick: PropTypes.func.isRequired,
};

export default Fireworks;
