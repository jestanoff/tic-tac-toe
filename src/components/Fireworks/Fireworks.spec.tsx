import * as React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import Fireworks from './index';

const emptyFn = () => undefined;

describe('<Fireworks />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should display third party fireworks animation when human player wins the game', () => {
    const { asFragment } = render(
      <div id="main-container">
        <Fireworks handleClick={emptyFn} />
      </div>,
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('should hide the execute the handleClick function when clicking on the canvas', async () => {
    const handleClickMock = jest.fn();
    render(
      <div id="main-container">
        <Fireworks handleClick={handleClickMock} />
      </div>,
    );

    const canvas = screen.getByRole('button');
    userEvent.click(canvas);

    await waitFor(() => expect(handleClickMock).toHaveBeenCalled());
    expect(handleClickMock).toHaveBeenCalledTimes(1);
    expect(handleClickMock).toHaveBeenCalledWith(expect.any(Object));
  });
});
