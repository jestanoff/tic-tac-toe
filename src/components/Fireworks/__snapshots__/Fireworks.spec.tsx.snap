// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Fireworks /> should display third party fireworks animation when human player wins the game 1`] = `
<DocumentFragment>
  <div
    id="main-container"
  >
    <canvas
      class="canvas"
      height="0"
      id="canvas"
      role="button"
      width="0"
    />
  </div>
</DocumentFragment>
`;
